api/client.go

```diff
@@ -1192,17 +1197,7 @@ START:
 	}
 	if err != nil {
 		if strings.Contains(err.Error(), "tls: oversized") {
-			err = errwrap.Wrapf(
-				"{{err}}\n\n"+
-					"This error usually means that the server is running with TLS disabled\n"+
-					"but the client is configured to use TLS. Please either enable TLS\n"+
-					"on the server or run the client with -address set to an address\n"+
-					"that uses the http protocol:\n\n"+
-					"    vault <command> -address http://<address>\n\n"+
-					"You can also set the VAULT_ADDR environment variable:\n\n\n"+
-					"    VAULT_ADDR=http://<address> vault <command>\n\n"+
-					"where <address> is replaced by the actual address to the server.",
-				err)
+			err = fmt.Errorf("%s\n\n%s", err, TLSErrorString)
```

Comment:
Switching from `errwrap` to `fmt.Errorf` might cause issues for the clients checking the inner (wrapped) error for equality (maybe not, I'm not really familiar with `errwrap`)

Reply:
If a client is currently using [this function](https://github.com/hashicorp/errwrap/blob/master/errwrap.go#L68-L73) to check for specific errors, it may start failing.

Reply:
Good call, I kept this as `errwrap` for backward compatibility.

Reply:
@ncabatoff I'd love to get your take on this. For this PR I think it's a good call to leave `errwrap` for these specific cases, but we can add something to our backlog that free's up the last few uses of `errwrap` in `client.go` if it's a reasonable proposition.